cdap-app-fabric/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -81,4 +127,205 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
     String uuid = bundleService.generateSupportBundle(bundleConfig);
     responder.sendString(HttpResponseStatus.OK, uuid);
   }
+
+  /**
+   * Get the list of bundle status from status.json.
+   *
+   * @throws IOException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundle")
+  public void getSupportBundle(HttpRequest request, HttpResponder responder) throws Exception {
+    /** ensure the user has authentication to create supportBundle */
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.GET);
+    File baseDirectory = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR));
+    if (!baseDirectory.exists()) {
+      responder.sendString(HttpResponseStatus.OK, "No content in Support Bundle.");
+      return;
+    }
+    JsonArray statusJsonArray = new JsonArray();
+    File[] supportFiles =
+      baseDirectory.listFiles((dir, name) -> !name.startsWith(".") && !dir.isHidden() && dir.isDirectory());
+    if (supportFiles != null && supportFiles.length > 0) {
+      for (File uuidFile : supportFiles) {
+        JsonObject singleBundleJson = bundleService.getSingleBundleJson(uuidFile);
+        statusJsonArray.add(singleBundleJson);
+      }
+    }
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(statusJsonArray));
+  }
+
+  /**
+   * Get the specific bundle status from status.json by using uuid.
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws IOException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundle/{uuid}")
+  public void getSupportBundleByUUID(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.GET);
+    File baseDirectory = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR));
+    File uuidFile = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR), uuid);
```

Comment:
This is probably not necessary; instead, consider passing only the UUID to the service. The service should understand the underlying implementation of the support bundle storage (i.e. it should know to concatenate the UUID with the storage directory path). The HTTP handler should not be aware of how implementation of the storage layer is done. This is applicable to all instances in which the handler calls the service.